core/templates/pages/exploration-editor-page/editor-tab/exploration-editor-tab.component.spec.ts

```diff
@@ -697,314 +714,124 @@ describe('Exploration editor tab component', function() {
+    const state = new State(
+      'stateName', 'id', 'some', null,
+      new Interaction([], [], null, null, [], 'id', null),
+      null, null, true, true, null, 7);
+    component.stateName = 'stateName';
+    spyOn(explorationStatesService, 'getState').and.returnValues(
+      state
     );
+    spyOn(explorationStatesService, 'isInitialized')
+      .and.returnValue(true);
+    spyOn(component, 'startTutorial');
+    editabilityService.onStartTutorial();
 
-    testsubscription.unsubscribe();
-  });
+    component.initStateEditor();
+    mockRefreshStateEditorEventEmitter.emit();
 
-  it('should start tutorial if in tutorial mode on page load', () => {
-    spyOn(ctrl, 'startTutorial');
-    editabilityService.onStartTutorial();
-    $scope.$apply();
-    ctrl.initStateEditor();
-    $scope.$apply();
-    expect(ctrl.startTutorial).toHaveBeenCalled();
+    expect(component.startTutorial).toHaveBeenCalled();
   });
 
   it('should check if exploration is editable', () => {
     spyOn(editabilityService, 'isEditable').and.returnValue(true);
-    expect(ctrl.isEditable()).toBe(true);
+    expect(component.isEditable()).toBe(true);
   });
 
   it('should not start tutorial if not in tutorial mode on page load', () => {
-    spyOn(ctrl, 'startTutorial');
+    spyOn(component, 'startTutorial');
     editabilityService.onEndTutorial();
-    $scope.$apply();
-    ctrl.initStateEditor();
-    $scope.$apply();
-    expect(ctrl.startTutorial).not.toHaveBeenCalled();
-  });
 
-  it('should finish tutorial if finish tutorial button is clicked', () => {
-    var registerFinishTutorialEventSpy = (
-      spyOn(siteAnalyticsService, 'registerFinishTutorialEvent'));
-    spyOn(editabilityService, 'onEndTutorial');
-    editabilityService.onStartTutorial();
-    $scope.$apply();
-    ctrl.initStateEditor();
-    $scope.$apply();
-    ctrl.onFinishTutorial();
-    expect(registerFinishTutorialEventSpy).toHaveBeenCalled();
-    expect(editabilityService.onEndTutorial).toHaveBeenCalled();
-    expect(ctrl.tutorialInProgress).toBe(false);
-  });
+    component.initStateEditor();
 
-  it('should skip tutorial if skip tutorial button is clicked', () => {
-    var registerSkipTutorialEventSpy = (
-      spyOn(siteAnalyticsService, 'registerSkipTutorialEvent'));
-    spyOn(editabilityService, 'onEndTutorial');
-    editabilityService.onStartTutorial();
-    $scope.$apply();
-    ctrl.initStateEditor();
-    $scope.$apply();
-    ctrl.onSkipTutorial();
-    expect(registerSkipTutorialEventSpy).toHaveBeenCalled();
-    expect(editabilityService.onEndTutorial).toHaveBeenCalled();
-    expect(ctrl.tutorialInProgress).toBe(false);
+    expect(component.startTutorial).not.toHaveBeenCalled();
   });
 
-  // The describe block below tests all the possible functions
-  // included on ctrl.EDITOR_TUTORIAL_OPTIONS array, which manipulates
-  // with JQuery the 'save from tutorial' button.
-  describe('when testing functions for JQuery manipulation from' +
```

Comment:
Why is this removed? Should there be equivalent tests added?

Reply:
i have replaced ng-joyride with ngx-joyride so this is no longer required,

in  ngx-joyride there is no as such testing required as everything is handled by the ngx-joyride package itself we have to only declare start and stop function apart from this everything is handle by the module itself.